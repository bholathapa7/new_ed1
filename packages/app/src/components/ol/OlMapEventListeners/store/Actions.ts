import { PostContentArguments } from '^/store/duck/Contents';
import * as T from '^/types';
import { Coordinate } from 'ol/coordinate';
import { Polygon } from 'ol/geom';

export enum ActionTypes {
  REGISTER_CLICK_EVENT = 'REGISTER_CLICK_EVENT',
  DEREGISTER_CLICK_EVENT = 'DEREGISTER_CLICK_EVENT',
  START_MODIFYING = 'START_MODIFYING',
  END_MODIFYING = 'END_MODIFYING',
  START_TRANSLATING = 'START_TRANSLATING',
  END_TRANSLATING = 'END_TRANSLATING',
  START_DRAWING = 'START_DRAWING',
  END_DRAWING = 'END_DRAWING',
  START_HOVERING_ON_LAYER = 'START_HOVERING_ON_LAYER',
  END_HOVERING_ON_LAYER = 'END_HOVERING_ON_LAYER',
  END_DRAWING_VOLUME = 'END_DRAWING_VOLUME',
  END_VOLUME_CREATION_PROCESS = 'END_VOLUME_CREATION_PROCESS',
  SELECT_LENGTH = 'SELECT_LENGTH',
  UPDATE_LENGTH_ELEVATIONS = 'UPDATE_LENGTH_ELEVATIONS',
  SHOW_LENGTH_SEGMENT_OVERLAYS = 'SHOW_LENGTH_SEGMENT_OVERLAYS',
  HIDE_LENGTH_SEGMENT_OVERLAYS = 'HIDE_LENGTH_SEGMENT_OVERLAYS',
  START_MODIFYING_VOLUME = 'START_MODIFYING_VOLUME',
  END_MODIFYING_VOLUME = 'END_MODIFYING_VOLUME',
  UPDATE_GEOLOCATION_POSITION = 'UPDATE_GEOLOCATION_POSITION',
  UPDATE_GEOLOCATION_ACCURACY_GEOMETRY = 'UPDATE_GEOLOCATION_ACCURACY_GEOMETRY',
  INITIAL_GEOLOCATION = 'INITIAL_GEOLOCATION',
}

export interface RegisterClickEvent {
  type: ActionTypes.REGISTER_CLICK_EVENT;
  payload: {
    currentlyModifiedLayerId?: number;
  };
}

export interface DeRegisterClickEvent {
  type: ActionTypes.DEREGISTER_CLICK_EVENT;
}

export interface StartDrawing {
  type: ActionTypes.START_DRAWING;
}
export interface EndDrawing {
  type: ActionTypes.END_DRAWING;
}

export interface StartTranslating {
  type: ActionTypes.START_TRANSLATING;
}
export interface EndTranslating {
  type: ActionTypes.END_TRANSLATING;
}

export interface StartModifying {
  type: ActionTypes.START_MODIFYING;
}

export interface EndModifying {
  type: ActionTypes.END_MODIFYING;
}
export interface StartHoveringOnLayer {
  type: ActionTypes.START_HOVERING_ON_LAYER;
  payload: { layerId: number };
}
export interface EndHoveringOnLayer {
  type: ActionTypes.END_HOVERING_ON_LAYER;
}

export interface EndDrawingVolume {
  type: ActionTypes.END_DRAWING_VOLUME;
  payload: Pick<T.VolumeContent, PostContentArguments>;
}

export interface EndVolumeCreationProcess {
  type: ActionTypes.END_VOLUME_CREATION_PROCESS;
}
export interface ShowLengthSegmentOverlays {
  type: ActionTypes.SHOW_LENGTH_SEGMENT_OVERLAYS;
}
export interface HideLengthSegmentOverlays {
  type: ActionTypes.HIDE_LENGTH_SEGMENT_OVERLAYS;
}

export interface DrawLength {
  type: ActionTypes.SELECT_LENGTH;
  payload?: {
    coordinates: Array<Coordinate>;
  };
}

export interface UpdateLengthElevations {
  type: ActionTypes.UPDATE_LENGTH_ELEVATIONS;
  payload: {
    elevations: Array<number>;
  };
}

export interface StartModifyingVolume {
  type: ActionTypes.START_MODIFYING_VOLUME;
}

export interface EndModifyingVolume {
  type: ActionTypes.END_MODIFYING_VOLUME;
}

export interface UpdateGeolocationPosition {
  type: ActionTypes.UPDATE_GEOLOCATION_POSITION;
  payload: { position: Coordinate };
}

export interface UpdateGeolocationAccuracyGeometry {
  type: ActionTypes.UPDATE_GEOLOCATION_ACCURACY_GEOMETRY;
  payload: { accuracyGeometry: Polygon };
}

export interface InitialGeolocation {
  type: ActionTypes.INITIAL_GEOLOCATION;
}

export type Actions =
  StartDrawing | EndDrawing | RegisterClickEvent | DeRegisterClickEvent |
  StartModifying | EndModifying | StartTranslating | EndTranslating | EndDrawingVolume |
  EndVolumeCreationProcess | DrawLength | UpdateLengthElevations | ShowLengthSegmentOverlays | HideLengthSegmentOverlays |
  StartHoveringOnLayer | EndHoveringOnLayer | StartModifyingVolume | EndModifyingVolume | UpdateGeolocationPosition |
  UpdateGeolocationAccuracyGeometry | InitialGeolocation;
