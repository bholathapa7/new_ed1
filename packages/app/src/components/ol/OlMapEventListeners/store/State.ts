import { PostContentArguments } from '^/store/duck/Contents';
import { Coordinate } from 'ol/coordinate';

import * as T from '^/types';
import { Polygon } from 'ol/geom';

export interface OlEventListenerState {
  isDrawing: boolean;
  showLengthSegmentOverlays: boolean;
  isClickEventRegistered: boolean;
  isModifying: boolean;
  isTranslating: boolean;
  isHoveringOnLayer: boolean;
  currentlyHoveredLayerId?: number;
  /**
   * We use this to track which layer is being modified.
   * Used for pointermove and modify.
   */
  currentlyModifiedLayerId?: number;
  /**
   * This will be defined after volume has been drawn
   */
  drawnVolume?: Pick<T.VolumeContent, PostContentArguments>;
  selectedLength?: { coordinates: Array<Coordinate> };
  selectedLengthElevations: { elevations: Array<number> };
  /**
   * This will monitor VC observable
   */
  isStillVolumeModifying: boolean;

  position?: Coordinate;
  accuracyGeometry?: Polygon;
}

export const initialState: OlEventListenerState = {
  isDrawing: false,
  showLengthSegmentOverlays: false,
  isModifying: false,
  isTranslating: false,
  isHoveringOnLayer: false,
  currentlyHoveredLayerId: undefined,
  currentlyModifiedLayerId: undefined,
  isClickEventRegistered: false,
  drawnVolume: undefined,
  selectedLength: undefined,
  selectedLengthElevations: { elevations: [] },
  isStillVolumeModifying: false,
  position: undefined,
  accuracyGeometry: undefined,
};
