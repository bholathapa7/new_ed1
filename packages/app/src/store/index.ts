import { routerMiddleware } from 'connected-react-router';
import { History } from 'history';
import _ from 'lodash-es';
import {
  Reducer, Store, StoreEnhancer,
  applyMiddleware, compose as defaultCompose, createStore,
} from 'redux';
import { batchedSubscribe } from 'redux-batched-subscribe';
import { composeWithDevTools } from 'redux-devtools-extension';
import { EpicMiddleware, createEpicMiddleware } from 'redux-observable';
import { persistReducer } from 'redux-persist';
import { BehaviorSubject } from 'rxjs';
import { switchMap } from 'rxjs/operators';

import config from '^/config';

import { EPIC_RELOAD } from '^/constants/epic';

import { State } from '^/types';
import createRootReducer, { RootAction, RootEpic, rootEpic } from './duck';
import persistConfig from './persistConfig';

const withDevtools: boolean = config.isNotProduction && config.isBrowser;

const compose: typeof defaultCompose = withDevtools ?
  composeWithDevTools({ maxAge: 500 }) :
  defaultCompose;

const epicMiddleware: EpicMiddleware<RootAction, RootAction, State> = createEpicMiddleware();

const debounceNotify: ((notify: Function) => void) = _.debounce((notify: Function) => notify());

const configStore: (
  history: History,
) => Store<State, RootAction> = (
  history,
) => {
  const enhancers: StoreEnhancer = compose(
    applyMiddleware(
      epicMiddleware,
      routerMiddleware(history),
    ) as StoreEnhancer,
    batchedSubscribe(debounceNotify),
  );

  const store: Store<State, RootAction> = createStore(
    persistReducer(
      persistConfig,
      createRootReducer(history),
    ) as Reducer<State, RootAction>,
    enhancers,
  );

  if (config.isNotProduction) {
    const epic$: BehaviorSubject<RootEpic> = new BehaviorSubject(rootEpic);
    const hotReloadingEpic: RootEpic = (
      action$, state$, dependencies,
    ) => epic$.pipe(
      switchMap((epic) => epic(action$, state$, dependencies)),
    );

    epicMiddleware.run(hotReloadingEpic);

    // eslint-disable-next-line @typescript-eslint/strict-boolean-expressions
    if (module.hot && module.hot.accept) {
      module.hot.accept('./duck', () => {
        store.replaceReducer(
          persistReducer(
            persistConfig,
            createRootReducer(history),
          ) as Reducer<State, RootAction>,
        );
        epic$.next(rootEpic);
        store.dispatch({ type: EPIC_RELOAD });
      });
    }
  } else {
    epicMiddleware.run(rootEpic);
  }

  return store;
};

export default configStore;
