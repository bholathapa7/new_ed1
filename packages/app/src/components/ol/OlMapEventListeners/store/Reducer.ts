import { exhaustiveCheck } from '^/utilities/exhaustive-check';
import { ActionTypes, Actions } from './Actions';
import { OlEventListenerState, initialState } from './State';

export function reducer(state: OlEventListenerState = initialState, action: Actions): OlEventListenerState {
  switch (action.type) {
    case ActionTypes.REGISTER_CLICK_EVENT:
      return {
        ...state,
        isClickEventRegistered: true,
        currentlyModifiedLayerId: action.payload.currentlyModifiedLayerId,
      };
    case ActionTypes.DEREGISTER_CLICK_EVENT:
      return {
        ...state,
        isClickEventRegistered: true,
        currentlyModifiedLayerId: undefined,
      };
    case ActionTypes.START_DRAWING:
      return { ...state, isDrawing: true };
    case ActionTypes.END_DRAWING:
      return { ...state, isDrawing: false };
    case ActionTypes.START_MODIFYING:
      return { ...state, isModifying: true };
    case ActionTypes.END_MODIFYING:
      return { ...state, isModifying: false };
    case ActionTypes.START_HOVERING_ON_LAYER:
      return { ...state, isHoveringOnLayer: true, currentlyHoveredLayerId: action.payload.layerId };
    case ActionTypes.END_HOVERING_ON_LAYER:
      return { ...state, isHoveringOnLayer: false, currentlyHoveredLayerId: undefined };
    case ActionTypes.START_TRANSLATING:
      return { ...state, isTranslating: true };
    case ActionTypes.END_TRANSLATING:
      return { ...state, isTranslating: false };
    case ActionTypes.END_DRAWING_VOLUME:
      return { ...state, drawnVolume: action.payload };
    case ActionTypes.END_VOLUME_CREATION_PROCESS:
      return { ...state, drawnVolume: undefined };
    case ActionTypes.SELECT_LENGTH:
      return { ...state, selectedLength: action.payload };
    case ActionTypes.UPDATE_LENGTH_ELEVATIONS:
      return { ...state, selectedLengthElevations: { elevations: action.payload.elevations } };
    case ActionTypes.SHOW_LENGTH_SEGMENT_OVERLAYS:
      return { ...state, showLengthSegmentOverlays: true };
    case ActionTypes.HIDE_LENGTH_SEGMENT_OVERLAYS:
      return { ...state, showLengthSegmentOverlays: false };
    case ActionTypes.START_MODIFYING_VOLUME:
      return { ...state, isStillVolumeModifying: true };
    case ActionTypes.END_MODIFYING_VOLUME:
      return { ...state, isStillVolumeModifying: false };
    case ActionTypes.UPDATE_GEOLOCATION_POSITION:
      return { ...state, position: action.payload.position };
    case ActionTypes.UPDATE_GEOLOCATION_ACCURACY_GEOMETRY:
      return { ...state, accuracyGeometry: action.payload.accuracyGeometry };
    case ActionTypes.INITIAL_GEOLOCATION:
      return { ...state, position: undefined, accuracyGeometry: undefined };
    default:
      exhaustiveCheck(action);
  }
}
