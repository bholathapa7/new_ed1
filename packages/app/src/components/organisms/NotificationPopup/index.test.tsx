import * as ReactTesting from '@testing-library/react';
import React, { createRef } from 'react';
import { Provider } from 'react-redux';
import createMockStore, { MockStore, MockStoreCreator } from 'redux-mock-store';

import * as M from '^/store/Mock';
import * as T from '^/types';

import { commonAfterEach } from '^/utilities/test-util';

import NoticePopup, { Props } from './index';

describe('NoticePopup', () => {
  type MockState = Pick<T.State, 'Pages' | 'Users' | 'PlanConfig'>;
  const mockStoreCreator: MockStoreCreator<MockState> = createMockStore<MockState>();
  const mockState: MockState = {
    Pages: {
      ...M.mockPages,
      Common: {
        ...M.mockPages.Common,
        language: T.Language.KO_KR,
      },
    },
    Users: {
      ...M.mockUsers,
      getNoticeStatus: T.APIStatus.SUCCESS,
    },
    PlanConfig: M.mockPlanConfig,
  };

  const createProps: () => Props = () => ({
    parentRef: createRef(),
    isNotificationPanelOpened: true,
    setIsNotificationPanelOpened: jest.fn(),
  });
  let store: MockStore<MockState>;
  let props: Props;
  let renderResult: ReactTesting.RenderResult;

  beforeEach(() => {
    store = mockStoreCreator(mockState);
    props = createProps();
    renderResult = ReactTesting.render(
      <Provider store={store}>
        <NoticePopup {...props} />
      </Provider>,
    );
  });

  afterEach(ReactTesting.cleanup);
  afterEach(commonAfterEach);

  const getNotificationItems: () => Array<HTMLElement> = () => renderResult.queryAllByTestId('notice-notification-item');
  const getNoNoticePopup: () => HTMLElement | null = () => renderResult.queryByTestId('no-notice-popup');

  it('should do nothing if api is not SUCCESS', () => {
    store = mockStoreCreator({
      ...mockState,
      Users: {
        ...mockState.Users,
        getNoticeStatus: T.APIStatus.PROGRESS,
      },
    });

    renderResult.rerender(
      <Provider store={store}>
        <NoticePopup {...props} isNotificationPanelOpened={true} />
      </Provider>,
    );

    expect(getNoNoticePopup()).toBe(null);
    expect(getNotificationItems().length).toBe(0);
  });

  it('should show unshown notifications', () => {
    expect(renderResult.asFragment()).toMatchSnapshot();

    store = mockStoreCreator({
      ...mockState,
      Users: {
        ...mockState.Users,
        notices: Object.values(mockState.Users.notices).map((notice: T.Notice) => ({
          ...notice, isShown: false,
        })),
      },
    });

    renderResult.rerender(
      <Provider store={store}>
        <NoticePopup {...props} />
      </Provider>,
    );

    expect(getNotificationItems().length).toBe(0);
  });

  it('should show unread notices', () => {
    renderResult.rerender(
      <Provider store={store}>
        <NoticePopup {...props} isNotificationPanelOpened={true} />
      </Provider>,
    );

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
