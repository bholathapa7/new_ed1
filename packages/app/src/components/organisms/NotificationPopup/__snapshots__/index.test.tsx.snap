// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticePopup should show unread notices 1`] = `
<DocumentFragment>
  .c4 {
  width: 347px;
  opacity: 0.95;
  border-radius: 5px;
  background-color: rgb(255,255,255);
  margin-left: 24.5px;
  margin-right: 24.5px;
  margin-bottom: 7.5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 {
  width: 23px;
  height: 17px;
}

.c5 {
  position: relative;
  width: 35px;
  height: 35px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 22px 1px 19.5px 22px;
  background-color: rgb(31,71,130);
  border-radius: 100%;
}

.c7 {
  padding-left: 17px;
  margin-top: 15.5px;
  width: 257px;
}

.c10 {
  width: 10px;
  height: 10px;
  margin-left: 5px;
  margin-bottom: 5px;
  border-radius: 100%;
  background-color: rgb(232,130,133);
}

.c8 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  color: rgb(102,102,102);
  line-height: 1.44;
  font-size: 14px;
  font-weight: 700;
}

.c12 {
  position: relative;
  width: 100%;
  text-align: left;
  line-height: 1.44;
}

.c13 {
  color: rgb(102,102,102);
  font-size: 12px;
}

.c14 {
  list-style-type: none;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 5px;
  margin-top: 6px;
  margin-bottom: 17px;
  cursor: pointer;
}

.c11 {
  width: 29px;
  height: 29px;
  border-radius: 100%;
  margin-left: 15px;
  margin-bottom: 6px;
  padding-right: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c11:hover {
  border-radius: 100%;
  background-color: rgb(198,217,235);
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 500;
  line-height: 1.44;
  color: rgb(43,72,119);
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-underline-offset: 1px;
  padding: 6px 10px 6px 18px;
  border-radius: 4px;
}

.c16:hover {
  background-color: rgb(233,233,233);
}

.c17 {
  margin-top: 3px;
  margin-left: 6px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 50px;
  right: 35px;
  margin-top: 7px;
  max-height: 360px;
  border-radius: 6px;
  -webkit-backdrop-filter: blur(6px);
  backdrop-filter: blur(6px);
  background-color: rgba(255,255,255,0.78);
  box-shadow: 0 1px 8px 0 rgba(0,0,0,0.2),0 3px 4px 0 rgba(0,0,0,0.18),0 3px 3px 0 rgba(0,0,0,0.16);
}

.c1 {
  overflow: auto;
}

.c1::-webkit-scrollbar {
  width: 8px;
}

.c1::-webkit-scrollbar-thumb {
  background: #7186A4;
  border-radius: 5px;
}

.c1::-webkit-scrollbar-thumb:hover {
  background: #2A5087;
}

.c2 {
  width: 100%;
  height: 100%;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 100%;
  padding-top: 22px;
  padding-bottom: 16px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
           
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg />
              </div>
            </div>
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  플랫폼 ver37.1.2 업데이트
                </p>
                <div
                  class="c10"
                  data-testid="notice-item-unread"
                />
                <span
                  class="c11"
                  data-testid="notice-item-hide"
                >
                  <svg />
                </span>
              </div>
              <div
                class="c12"
              >
                <div
                  class="c13"
                >
                  플랫폼이 ver37.1.2으로 업데이트되었습니다.
                </div>
                <ul
                  class="c14"
                >
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    1. 오늘은 2019년 10월 28일
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    2. 내일은 2019년 10월 29일
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    3. 좀 있으면 11월
                  </li>
                </ul>
                <div
                  class="c15"
                >
                  <div
                    class="c16"
                  >
                    더 알아보기
                    <span
                      class="c17"
                    >
                      <svg />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg />
              </div>
            </div>
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  플랫폼 ver29.2.7 업데이트
                </p>
                <div
                  class="c10"
                  data-testid="notice-item-unread"
                />
                <span
                  class="c11"
                  data-testid="notice-item-hide"
                >
                  <svg />
                </span>
              </div>
              <div
                class="c12"
              >
                <div
                  class="c13"
                >
                  플랫폼이 ver29.2.7으로 업데이트되었습니다.
                </div>
                <ul
                  class="c14"
                >
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    1. 고양이
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    2. 강아지
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    3. 얼룩말
                  </li>
                </ul>
                <div
                  class="c15"
                >
                  <div
                    class="c16"
                  >
                    더 알아보기
                    <span
                      class="c17"
                    >
                      <svg />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg />
              </div>
            </div>
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  플랫폼 ver16.8.1 업데이트
                </p>
                <span
                  class="c11"
                  data-testid="notice-item-hide"
                >
                  <svg />
                </span>
              </div>
              <div
                class="c12"
              >
                <div
                  class="c13"
                >
                  플랫폼이 ver16.8.1으로 업데이트되었습니다.
                </div>
                <ul
                  class="c14"
                >
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    1. 시간여행 가능
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    2. 드론 자가용으로 사용 가능
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    3. 엔젤스윙 흑백모드 출시
                  </li>
                </ul>
                <div
                  class="c15"
                >
                  <div
                    class="c16"
                  >
                    더 알아보기
                    <span
                      class="c17"
                    >
                      <svg />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`NoticePopup should show unshown notifications 1`] = `
<DocumentFragment>
  .c4 {
  width: 347px;
  opacity: 0.95;
  border-radius: 5px;
  background-color: rgb(255,255,255);
  margin-left: 24.5px;
  margin-right: 24.5px;
  margin-bottom: 7.5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c6 {
  width: 23px;
  height: 17px;
}

.c5 {
  position: relative;
  width: 35px;
  height: 35px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 22px 1px 19.5px 22px;
  background-color: rgb(31,71,130);
  border-radius: 100%;
}

.c7 {
  padding-left: 17px;
  margin-top: 15.5px;
  width: 257px;
}

.c10 {
  width: 10px;
  height: 10px;
  margin-left: 5px;
  margin-bottom: 5px;
  border-radius: 100%;
  background-color: rgb(232,130,133);
}

.c8 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  color: rgb(102,102,102);
  line-height: 1.44;
  font-size: 14px;
  font-weight: 700;
}

.c12 {
  position: relative;
  width: 100%;
  text-align: left;
  line-height: 1.44;
}

.c13 {
  color: rgb(102,102,102);
  font-size: 12px;
}

.c14 {
  list-style-type: none;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-right: 5px;
  margin-top: 6px;
  margin-bottom: 17px;
  cursor: pointer;
}

.c11 {
  width: 29px;
  height: 29px;
  border-radius: 100%;
  margin-left: 15px;
  margin-bottom: 6px;
  padding-right: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c11:hover {
  border-radius: 100%;
  background-color: rgb(198,217,235);
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 12px;
  font-weight: 500;
  line-height: 1.44;
  color: rgb(43,72,119);
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-underline-offset: 1px;
  padding: 6px 10px 6px 18px;
  border-radius: 4px;
}

.c16:hover {
  background-color: rgb(233,233,233);
}

.c17 {
  margin-top: 3px;
  margin-left: 6px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  top: 50px;
  right: 35px;
  margin-top: 7px;
  max-height: 360px;
  border-radius: 6px;
  -webkit-backdrop-filter: blur(6px);
  backdrop-filter: blur(6px);
  background-color: rgba(255,255,255,0.78);
  box-shadow: 0 1px 8px 0 rgba(0,0,0,0.2),0 3px 4px 0 rgba(0,0,0,0.18),0 3px 3px 0 rgba(0,0,0,0.16);
}

.c1 {
  overflow: auto;
}

.c1::-webkit-scrollbar {
  width: 8px;
}

.c1::-webkit-scrollbar-thumb {
  background: #7186A4;
  border-radius: 5px;
}

.c1::-webkit-scrollbar-thumb:hover {
  background: #2A5087;
}

.c2 {
  width: 100%;
  height: 100%;
  border-radius: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 100%;
  padding-top: 22px;
  padding-bottom: 16px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
           
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg />
              </div>
            </div>
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  플랫폼 ver37.1.2 업데이트
                </p>
                <div
                  class="c10"
                  data-testid="notice-item-unread"
                />
                <span
                  class="c11"
                  data-testid="notice-item-hide"
                >
                  <svg />
                </span>
              </div>
              <div
                class="c12"
              >
                <div
                  class="c13"
                >
                  플랫폼이 ver37.1.2으로 업데이트되었습니다.
                </div>
                <ul
                  class="c14"
                >
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    1. 오늘은 2019년 10월 28일
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    2. 내일은 2019년 10월 29일
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    3. 좀 있으면 11월
                  </li>
                </ul>
                <div
                  class="c15"
                >
                  <div
                    class="c16"
                  >
                    더 알아보기
                    <span
                      class="c17"
                    >
                      <svg />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg />
              </div>
            </div>
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  플랫폼 ver29.2.7 업데이트
                </p>
                <div
                  class="c10"
                  data-testid="notice-item-unread"
                />
                <span
                  class="c11"
                  data-testid="notice-item-hide"
                >
                  <svg />
                </span>
              </div>
              <div
                class="c12"
              >
                <div
                  class="c13"
                >
                  플랫폼이 ver29.2.7으로 업데이트되었습니다.
                </div>
                <ul
                  class="c14"
                >
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    1. 고양이
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    2. 강아지
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    3. 얼룩말
                  </li>
                </ul>
                <div
                  class="c15"
                >
                  <div
                    class="c16"
                  >
                    더 알아보기
                    <span
                      class="c17"
                    >
                      <svg />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <div
                class="c6"
              >
                <svg />
              </div>
            </div>
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <p
                  class="c9"
                >
                  플랫폼 ver16.8.1 업데이트
                </p>
                <span
                  class="c11"
                  data-testid="notice-item-hide"
                >
                  <svg />
                </span>
              </div>
              <div
                class="c12"
              >
                <div
                  class="c13"
                >
                  플랫폼이 ver16.8.1으로 업데이트되었습니다.
                </div>
                <ul
                  class="c14"
                >
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    1. 시간여행 가능
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    2. 드론 자가용으로 사용 가능
                  </li>
                  <li
                    class="c13"
                    data-testid="notice-item-content"
                  >
                    3. 엔젤스윙 흑백모드 출시
                  </li>
                </ul>
                <div
                  class="c15"
                >
                  <div
                    class="c16"
                  >
                    더 알아보기
                    <span
                      class="c17"
                    >
                      <svg />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
